department-of-veterans-affairs/vets-website
# Enforce LOA3 before redirecting from "/sign-in"

## Description

For users attempting to navigate to My VA Health via the `/sign-in` page and `?application=myvahealth` parameters, these changes make sure that they are verified before redirecting externally. Otherwise, it prompts the user to verify their identity first. This necessitates some extra logic in `/auth/logic/callback`.

This grew the complexity of the sign-in page a little bit, so this PR also includes some restructuring of the app to be a little clearer in some logical divisions, namely breaking out a few sections of markup into the following files:

- `FedWarning.jsx`
- `SignInButtons.jsx`
- `SignInDescription.jsx`

Their content/functionality remains the same, but breaking them out helped keep `SignInApp.jsx` less bloated.

Closes department-of-veterans-affairs/va.gov-team#11292


## Testing done

Manual verification of these scenarios:

- LOA1 user logs in via `/sign-in/?application=myvahealth`, gets prompted to verify after logging in
- LOA1 user who is logged in navigates to `/sign-in/?application=myvahealth`, gets prompted to verify
- LOA3 user logs in via `/sign-in/?application=myvahealth`, gets redirected to My VA Health after successful login

Also updated unit tests

## Screenshots

![Screenshot_2020-07-31 Login Page](https://user-images.githubusercontent.com/13280995/89093015-3661b580-d37c-11ea-9510-56645298e2be.png)

## Acceptance criteria
- [ ] Unverified users navigating to My VA Health via `/sign-in` are forced to verify their identity before being redirected

## Definition of done
